### PR TITLE
Show neuroglancer meshes in webKnossos

### PR DESCRIPTION
**Pull request by normanrz**
This PR adds support for fetching/parsing meshes in the [neuroglancer precomputed format](https://github.com/google/neuroglancer/blob/master/src/neuroglancer/datasource/precomputed/meshes.md) and making it available for webKnossos using the standard mesh API. Since webKnossos doesn't support level-of-detail yet, the lod is hard-coded to 2 :D

### CODE REVIEW

File: wkconnect/backends/neuroglancer/backend.py [hunk 1]
```diff
@@ -358,7 +380,101 @@ async def read_data(
...
+    async def __read_meshfile(
+        self, layer: Layer, segment_id: int, token: Optional[Token]
+    ) -> Optional[Meshfile]:
+        assert layer.mesh is not None
+        mesh_info = layer.mesh
+        sharding_info = mesh_info.sharding
+        chunk_id = np.uint64(segment_id)
+        minishard_info = sharding_info.get_minishard_info(chunk_id)
+        shard_url = f"{layer.source}/mesh/{sharding_info.format_shard_for_url(minishard_info)}.shard"
+        # print("shard_url", shard_url)
```
Comment (fm3) [inline wkconnect/backends/neuroglancer/backend.py:393]: prints can be removed

File: wkconnect/backends/neuroglancer/meshes.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+            _transform=tuple(info_json["transform"]),
+            vertex_quantization_bits=info_json["vertex_quantization_bits"],
+            lod_scale_multiplier=info_json["lod_scale_multiplier"],
+        )
+
+    @property
+    def transform(self) -> np.ndarray:
+        return np.array(self._transform, dtype="f4").reshape((3, 4))
+
+    def parse_meshfile(self, buf: bytes, byte_offset: int) -> Meshfile:
```
Comment (fm3) [inline wkconnect/backends/neuroglancer/meshes.py:84]: I guess this method is based on the documentation of the format? Is it this? https://github.com/google/neuroglancer/blob/master/src/neuroglancer/datasource/precomputed/meshes.md ? Could we link it here in a comment?

File: wkconnect/backends/neuroglancer/sharding.py [hunk 3]
```diff
@@ -103,6 +101,25 @@ def __post_init__(self) -> None:
             self, "_minishard_mask", compute_minishard_mask(self.minishard_bits)
         )
 
+    @staticmethod
+    def parse(
+        info_json: Any,
+        dataset_size: Optional[Vec3D] = None,
+        chunk_size: Optional[Vec3D] = None,
+    ) -> "ShardingInfo":
+        assert info_json["@type"] == "neuroglancer_uint64_sharded_v1"
```
Comment (fm3) [inline wkconnect/backends/neuroglancer/sharding.py:110]: What happens if the assertions are not green? Is this propagated to the user? Should there be readable assertion messages?

File: wkconnect/backends/neuroglancer/sharding.py [hunk 4]
```diff
@@ -114,10 +131,14 @@ def get_chunk_key(self, pos: Vec3D) -> np.uint64:
     def format_shard_for_url(self, loc: MinishardInfo) -> str:
         return format(loc.shard_number, "x").zfill(int(np.ceil(self.shard_bits / 4.0)))
 
+    def hash_chunk_id(self, chunk_id: np.uint64) -> np.uint64:
+        if self.hashfn == "murmurhash3_x86_128":
+            return np.uint64(hash64(chunk_id.tobytes())[0])
+        return chunk_id
```
Comment (fm3) [inline wkconnect/backends/neuroglancer/sharding.py:137]: should this throw an error if a different hashfn is passed?

File: wkconnect/routes/datasets/meshes.py [hunk 5]
```diff
@@ -10,7 +12,66 @@
...
+        return response.json(chunks)
+    except NotImplementedError:
+        return response.empty(status=501)
+
+
+@meshes.route(
+    "/<organization_name>/<dataset_name>/layers/<layer_name>/meshes/chunks/data",
+    methods=["POST"],
+)
+async def get_mesh_chunk_data(
```
Comment (fm3) [inline wkconnect/routes/datasets/meshes.py:56]: the routes may need `@authorized(AccessRequest.read_dataset)` so the authentication is checked

**Review by fm3 — COMMENTED** [review]
Code mostly LGTM, see my small comments. I will admit that I did not go throught he whole documentation and buffer parsing code line by line. Is there some way for me to test this?

**Comment by normanrz** [comment]
> Is there some way for me to test this?

You could start wkconnect on your dev setup and use the test dataset from the default datasets.json: https://github.com/scalableminds/webknossos-connect/blob/neuroglancer-meshes/data/datasets.json#L21

**Review by fm3 — APPROVED** [review]
Cool stuff! I think I may have already found a merge error in the test dataset by looking at the mesh :D LGTM

**Pull request merged**